application/cms/views.py

```diff
@@ -1109,3 +1112,95 @@ def view_measure_version_by_measure_version_id(measure_version_id):
             version=measure_version.version,
         )
     )
+
+
+@cms_blueprint.route("/data-sources/new", methods=["GET"])
+@login_required
+def new_data_source():
+
+    data_source_form = DataSourceForm()
+    organisations_by_type = Organisation.select_options_by_type()
+
+    return render_template(
+        "cms/new_data_source.html", data_source_form=data_source_form, organisations_by_type=organisations_by_type
+    )
+
+
+@cms_blueprint.route("/data-sources", methods=["POST"])
+@login_required
+def create_data_source():
+
+    data_source_form = DataSourceForm(request.form)
+
+    data_source = DataSource()
+    data_source_form.populate_obj(data_source)
+
+    if data_source.title:
```

Comment:
We are skipping form validation here - maybe look at other views to see how they do it. But it should generally be something like `if form.validate_on_submit()`. This will then populate errors, which can be retrieved for the template with `errors=get_form_errors(forms=(form, ))`.

Reply:
Done